frontend_csv/src/scanner/scanner.rs

```diff
@@ -1,159 +1,189 @@
-                }
-                "u32" => {
-                    tokenize.push_back(CSVToken {
-                        _type: CSVTokenType::U32,
-                        _value: current_buffer.clone(),
-                    });
-                    current_buffer = String::new();
-                }
-                "u64" => {
-                    tokenize.push_back(CSVToken {
-                        _type: CSVTokenType::U64,
-                        _value: current_buffer.clone(),
-                    });
-                    current_buffer = String::new();
-                }
-                "channel_id" => {
-                    tokenize.push_back(CSVToken {
-                        _type: CSVTokenType::ChannelId,
-                        _value: current_buffer.clone(),
-                    });
-                    current_buffer = String::new();
-                }
-                "signature" => {
-                    tokenize.push_back(CSVToken {
-                        _type: CSVTokenType::Signature,
-                        _value: current_buffer.clone(),
-                    });
-                    current_buffer = String::new();
-                }
-                "point" => {
-                    tokenize.push_back(CSVToken {
-                        _type: CSVTokenType::Point,
-                        _value: current_buffer.clone(),
-                    });
-                    current_buffer = String::new();
-                }
-                "chain_hash" => {
-                    tokenize.push_back(CSVToken {
-                        _type: CSVTokenType::ChainHash,
-                        _value: current_buffer.clone(),
-                    });
-                    current_buffer = String::new();
-                }
-                "byte" => {
-                    tokenize.push_back(CSVToken {
-                        _type: CSVTokenType::Byte,
-                        _value: current_buffer.clone(),
-                    });
-                    current_buffer = String::new();
-                }
-                "bigsize" => {
-                    tokenize.push_back(CSVToken {
-                        _type: CSVTokenType::BigSize,
-                        _value: current_buffer.clone(),
-                    });
-                    current_buffer = String::new();
-                }
-                // "" => tokenToAdd = CSVToken{_type: CSVTokenType::, _value: current_buffer.clone()},
-                &_ => {
-                    assert_eq!(1, 1);
-                }
-            }
+    pub fn new(line: u64) -> Scanner {
+        let keywords = HashMap::from([
+            (
+                "msgtype".to_string(),
+                CSVToken {
+                    ty: CSVTokenType::MsgTy,
+                    val: "msgtype".to_string(),
+                },
+            ),
+            (
+                "msgdata".to_string(),
+                CSVToken {
+                    ty: CSVTokenType::MsgTy,
```

Comment:
```suggestion
                    ty: CSVTokenType::MsgData,
```

If your test does not fail this means that the scanner does not work, in fact, some parts are unclear to me.

Reply:
which part do you specific refer to?

Reply:
it is not a `MsgTyp` token but `MsgData` and  it was not check by the CI

Reply:
Hi I just check locally and the test failed as intend... so could it be the CI not reading the CSV file?

Reply:
I do not know, also in this case we should fail make a double check

Reply:
the test did not run cos in github action its cargo test --doc instead of cargo test...